examples/jsm/renderers/nodes/lights/LightContextNode.js

```diff
@@ -0,0 +1,36 @@
+import ContextNode from '../core/ContextNode.js';
+import { RE_Direct_BlinnPhong } from '../functions/BSDFs.js';
+
+class LightContextNode extends ContextNode {
+
+	constructor( node ) {
+
+		super( node );
+		
+	}
+
+	generate( builder, output ) {
+		
+		const material = builder.material;
+
+		let RE_Direct = null;
+		
+		if ( material.isMeshPhongMaterial === true ) {
+			
+			RE_Direct = RE_Direct_BlinnPhong;
```

Comment:
I have not yet debugged this part in detail but I assume the define `RE_Direct` in the phong shader will be overwritten by the generated code of `RE_Direct_BlinnPhong`?

Reply:
Yes, but `RE_Direct` is a `FunctionCallNode`. I am preserving the names to facilitate. `NodeMaterial` not need set a `define` for this. With `LightContextNode` and no `defines` we can adopt event two light model, like `Phong` and `Physical` runing simultaneously on the same material.